src/main/groovy/life/qbic/db/MariaDBManager.groovy

```diff
@@ -423,19 +424,25 @@ class MariaDBManager implements IQueryService, INotificationService {
       Location currLoc = null;
       java.util.Date currDate = null
       for(GroovyRowResult rs: results) {
-        int currID = rs.get("current_location_id")
-        int locID = rs.get("location_id")
+        int currID = rs.get("current_location_id") as int
+        int locID = rs.get("location_id") as int
         java.util.Date arrivalDate = toDate(rs.get("arrival_time"))
         java.util.Date forwardedDate = toDate(rs.get("forwarded_time"))
-        Status status = rs.get("sample_status")
+        Status status = rs.get("sample_status") as Status
         String name = rs.get("name")
         String street = rs.get("street")
         String country = rs.get("country")
-        int zip = rs.get("zip_code")
+        int zip = rs.get("zip_code") as int
 
         Address address = new Address(affiliation: name, street: street, zipCode: zip, country: country)
-        int personID = rs.get("responsible_person_id");
-        Person pers = getPersonNameByID(personID)
+        int personID = rs.get("responsible_person_id") as int
+        Person pers
+        try {
+          pers = getPersonByDbId(personID)
+        } catch (NotFoundException personNotFoundException) {
+          log.error("Could not retrieve responsible person for change {sample: ${code}; time of change: ${arrivalDate.toString()}}.", personNotFoundException)
+          throw new RuntimeException("The request for sample ${code} failed.")
+        }
 
         if(currID == locID ) {
```

Comment:
Maybe it makes sense to move all of that code into a separate method (NIT)

Reply:
Like the code below that tests for the correct sample location

Reply:
I would like to keep this as is for the moment. This would be a bigger refactor as the code is not easily separated (e.g. the extracted methods would have more than 5 parameters and would not be readable at all)